lib/src/Dscp/Educator/Block/Type.hs

```diff
@@ -0,0 +1,82 @@
+
+-- | Blocks- and blockchain-related datatypes definition.
+
+module Dscp.Educator.Block.Type
+       ( -- * Basic types
+         PrivateHeaderHash
+       , PrivateBlockHeader (..)
+       , pbhPrevBlock
+       , pbhBodyProof
+       , pbhAtgDelta
+       , PrivateBlockBody (..)
+       , pbbTxs
+       , PrivateBlock (..)
+       , pbHeader
+       , pbBody
+
+         -- * Constants
+       , genesisHeaderHash
+
+         -- * Helpers
+       , getPrevBlockRefMaybe
+       ) where
+
+import Control.Lens (makeLenses)
+import Dscp.Core.Types (ATGDelta (..))
+import Dscp.Crypto (Hash, unsafeHash)
+import Dscp.Crypto.MerkleTree (MerkleSignature)
+import Dscp.Educator.Txs (PrivateTx)
+
+----------------------------------------------------------
+-- Block elements
+----------------------------------------------------------
+
+-- | Hash of the private block.
+type PrivateHeaderHash = Hash PrivateBlockHeader
+
+-- | Header of a private block. There's no signatures here, as it's private anyway.
+-- During publishing, Educator will provide a signature as a part of transaction.
+data PrivateBlockHeader = PrivateBlockHeader
+    { _pbhPrevBlock :: !PrivateHeaderHash
+    -- ^ Previous header in the chain
+    , _pbhBodyProof :: !(MerkleSignature PrivateTx)
+    -- ^ Body payload proof (for now - only root of sized Merkle tree
+    -- over private transactions)
+    , _pbhAtgDelta  :: !ATGDelta
```

Comment:
Can you comment it out for now? We're not yet sure how to resolve it in public chain.